vim-patch:9.1.{1121,1126}

glepnir opened the pull request:
#### vim-patch:9.1.1121: Enter does not insert newline with "noselect"

Problem:  Enter does not insert newline with "noselect" when the pum is
          visible (lifepillar)
Solution: When Enter is pressed and no complete-item is selected,
          ins_compl_prep returns false, and the edit function continues
          processing Enter to insert a new line. (glepnir)

fixes: vim/vim#1653
closes: vim/vim#16653

https://github.com/vim/vim/commit/07f0dbe3aa326fdf4d0f1b1cf7d79df89e91fc6e


#### vim-patch:9.1.1126: patch 9.1.1121 used a wrong way to handle enter

Problem:  patch 9.1.1121 used a wrong way to handle enter
Solution: compl_enter_selects also needs to consider the selected item
          in ins_compl_new_leader() (glepnir)

closes: vim/vim#16673

https://github.com/vim/vim/commit/44180416981000ad0bc5db4686889892e7a05cdd

glepnir committed:
vim-patch:9.1.1121: Enter does not insert newline with "noselect"

Problem:  Enter does not insert newline with "noselect" when the pum is
          visible (lifepillar)
Solution: When Enter is pressed and no complete-item is selected,
          ins_compl_prep returns false, and the edit function continues
          processing Enter to insert a new line. (glepnir)

fixes: vim/vim#1653
closes: vim/vim#16653

https://github.com/vim/vim/commit/07f0dbe3aa326fdf4d0f1b1cf7d79df89e91fc6e

Co-authored-by: glepnir <glephunter@gmail.com>

zeertzjq commented:
Test failure

zeertzjq commented on test/functional/editor/completion_spec.lua:
```diff
@@ -512,13 +512,15 @@ describe('completion', function()
       poke_eventloop()
       feed('<CR>')
-      expect('Ju')
+      expect([[Ju
```
Test description needs change

zeertzjq commented on test/functional/editor/completion_spec.lua:
```diff
@@ -506,19 +506,21 @@ describe('completion', function()
     end)
 
-    it('Enter selects original text after adding leader', function()
+    it('Enter selects original text after adding leader and insert newline', function()
```
```suggestion
    it('Enter does not select original text after adding leader', function()
```

zeertzjq committed:
test(completion_spec): make Enter test descriptions more accurate

zeertzjq committed:
vim-patch:9.1.1126: patch 9.1.1121 used a wrong way to handle enter

Problem:  patch 9.1.1121 used a wrong way to handle enter
Solution: compl_enter_selects also needs to consider the selected item
          in ins_compl_new_leader() (glepnir)

closes: vim/vim#16673

https://github.com/vim/vim/commit/44180416981000ad0bc5db4686889892e7a05cdd

Co-authored-by: glepnir <glephunter@gmail.com>
